shared/utility/actuation/InverseKinematics.hpp

```diff
@@ -42,177 +42,143 @@ namespace utility::actuation::kinematics {
     using utility::input::LimbID;
     using utility::input::ServoID;
 
-    /*! @brief Calculates the leg joints for a given input ankle position.
-            The robot coordinate system has origin a distance DISTANCE_FROM_BODY_TO_HIP_JOINT above the midpoint
-       of the hips. Robot coordinate system: x is out of the front of the robot y is left, from right shoulder
-       to left z is upward, from feet to head Input ankle coordinate system: x is forward, from heel to toe y is
-       left, z is normal to the plane of the foot
-        @param target The target 4x4 basis matrix for the ankle
-        @param isLeft Request for left leg motors or right leg motors?
-        @param RobotKinematicsModel The class containing the leg model of the robot.
-    */
+    /** @brief Calculates the leg joints for a given input ankle position. The robot coordinate system has origin a
+     * distance DISTANCE_FROM_BODY_TO_HIP_JOINT above the midpoint of the hips. Robot coordinate system: x is out of the
+     * front of the robot y is left, from right shoulder to left z is upward, from feet to head Input ankle coordinate
+     * system: x is forward, from heel to toe y is left, z is normal to the plane of the foot
+     * @param model KinematicsModel containing dimensions of the robot
+     * @param Htf_ Desired pose of foot {f} in torso {t} space
+     * @param limb Which leg to solve IK for (left or right)
+     * @return Vector of pairs of servo ID and joint angles
+     */
     template <typename Scalar>
-    [[nodiscard]] std::vector<std::pair<ServoID, Scalar>> calculateLegJoints(
+    [[nodiscard]] std::vector<std::pair<ServoID, Scalar>> calculate_leg_joints(
         const message::actuation::KinematicsModel& model,
-        const Eigen::Transform<Scalar, 3, Eigen::Isometry>& target_,
+        const Eigen::Transform<Scalar, 3, Eigen::Isometry>& Htf_,
```

Comment:
If you're committed to using doubles everywhere (which is probably a good idea) you could replace all this math with a non template double version. That'd get rid of a lot of the dependent templates and let you use Eigen::Isometry3d and Eigen::Vector3d to make the later code easier to read

Reply:
I would prefer utilities to be templated for generality. I guess in this case the implementation is fairly restricted to a single use case but I like the convention of utilities always being templated